### PR TITLE
Update array_utils.py

### PR DESCRIPTION
**Pull request by averyma**
Increase tolerance from 1e-5 (default) to 1e-4 when checking sum_to_one

## Summary:
The PR relaxes the tolerance used to check whether an input vector is a valid probability distribution. The current check uses [np.allclose](https://numpy.org/doc/stable/reference/generated/numpy.allclose.html)(probabilities.sum(1), 1), which relies on the default rtol=1e-5. Vectors produced by torch.nn.Softmax in float32 often sum to around 1 ± 1e-4, so valid probability vectors are incorrectly rejected.

### Features:
To avoid complicating the existing code, this pr simply increases the tolerance to 1e-4 so that softmax outputs are correctly treated as probability distributions.

## Checklist:

* [x ] Your code is documented (To validate this, add your module to `tests/documentation_test.py`).
* [x ] Your code is tested with unit tests.
* [x ] You moved your Issue to the PR state.

### CODE REVIEW

File: baal/utils/array_utils.py [hunk 1]
```diff
@@ -15,7 +15,7 @@ def to_prob(probabilities: np.ndarray):
     """
     not_bounded = np.min(probabilities) < 0 or np.max(probabilities) > 1.0
     multiclass = probabilities.shape[1] > 1
-    sum_to_one = np.allclose(probabilities.sum(1), 1)
+    sum_to_one = np.allclose(probabilities.sum(1), 1, rtol=1e-4)
```
Comment (Dref360) [inline baal/utils/array_utils.py:18]: Based on your explanation, should this uses `atol` instead of `rtol`? I guess in this case since `b==1`, it doesn't matter.

Comment (averyma) [reply baal/utils/array_utils.py:18]: Thank you for the response. You are correct it does not matter in this case.

**Review by Dref360 — APPROVED** [review]
LGTM Thank you!
